### PR TITLE
Separate SDK Linux shards into separate builders - prod

### PR DESCRIPTION
**Pull request by keyonghan**
This create separate builder configs for each subshard (Linux prod ones).

Related: flutter/flutter#77947

### CODE REVIEW

File: config/lib/common.star [hunk 1]
```diff
@@ -442,6 +444,13 @@ def _builder_with_subshards(
                 os = os,
                 **kwargs
             )
+        elif bucket == "prod" and os.startswith("Linux"):
```
Comment (godofredoc) [inline config/lib/common.star:447]: if we pass the name can we delete this branch of the code?

Comment (keyonghan) [reply config/lib/common.star:447]: Actually the `name` is not needed (skipped from the framework.star side), making it easy here to generate the name based on branch, shards, and subshard. 
(the `name` is also not needed for try builder configs, I plan to remove it in a following PR)

Comment (godofredoc) [reply config/lib/common.star:447]: What will be the name if branch_postfix is None? e.g. we are not adding master to the name. Additionally can we call the parameter just branch? it is misleading to use branch_postfix in the middle of the string.

Comment (keyonghan) [reply config/lib/common.star:447]: Force inject the parameters. Since `branch` variable already exists, used `branch_name` instead.

**Review by godofredoc — APPROVED** [review]

**Pull request merged**